vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+Multi-theading allows for the calculations to be spread across multiple threads, which could accelerate inference by allowing embarassingly parallel calculations to be spread across threads and then later combined.
+If possible, a good default is `2`, however, if you have available compute on your machine, you can increase this to a higher value and may see some fit time reductions.
+Please note that the `theads_per_chain` has not impact on model convergence or fittings and is purely a means of speeding up the model fitting time.
+
+::: {.alert .alert-warning}
+**The product of the `thread_per_chain` and `chains`** arguments should be equal to or less than the number of cores on your machine! 
+:::
+
+### iter_warmup and iter_sampling {.iters}
+
+The number of samples using for the warmup phase of Stan and the number of sampling iterations are controlled by the `iter_warmup` and `iter_sampling` arguments, respectively.
+Stan requires a sufficient number of warmup iterations to dynamically estimate the step change sizes during model fitting. 
+Unlike with Gibbs sampling (such as the method used in JAGS), you do not need to set large values for `iter_warmup` or `iter_sampling`.
+Conversely, setting large values for iterations is a sign of poor model fit and indicates that other approaches need to be employed (e.g., `priors`, `adapt_delta`, `max_treedepth`)
+Typically, 1000-2000 samples for `iter_warmup` and `iter_sampling` are sufficient.
+
+### max_treedepth {.maxTree}
+
+This value controls the size of the trajectory taken by the sample (in power of 2) take for each step.
+If this value is too low, the sampler may terminate too early causing excess runtimes.
+You can increase the `max_treedepth` to an integer value greater than `10` up to `15` (or lower).
+Given the complexity of the models used in `{epinowcast}` it is common to have the `max_treedepth` set between 12 and 15.
+
+### adapt_delta {.adaptDelta}
+
+This parameter sets the target average proposal acceptable probability during NUTS sampling (and is a real number values between 0 and 1).
+Higher values of `adapt_delta` will result in smaller step sizes during sampling which will slow the model fitting time but can help to address *divergent transitions.*
+The default value is 0.8, but it is likely that this will need to be set slightly higher like 0.9 to 0.99 given the complexity of the models being fit.
+
+### Some decent defaults
+
+For a computer with 8 cores, a reasonable configuration would be the following:
+
+```r
+enw_fit_opts(
+    save_warmup = FALSE,
+    pp = TRUE,
+    chains = 4,
+    threads_per_chain = 2,
+    adapt_delta = 0.95,
+    max_treedepth = 12,
+    iter_sampling = 2000,
+    iter_warmup = 1000,
+    show_messages = FALSE
+)
+```
+
+## Model settings {.modelParams}
+
+### Setting priors {.settingPriors}
+
+Setting sensible priors is important for Bayesian inference.
```

Comment:
This may be beyond the scope of this doc, but when investigating divergent transitions I've found it helpful to turn `adapt_delta` back down to 0.8. It gives a much better view of the divergences and can help make clearer where the problem is coming from. It's probably also good to throw another cite here to the Bayesian workflow paper and mention prior predictive sim?

Reply:
> when investigating divergent transitions I've found it helpful to turn adapt_delta back down to 0.8

Yes agree and agree on citing out to the workflow paper liberally.